subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_df.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <errno.h>
+#include <devicetree.h>
+#include <sys/util_macro.h>
+#include <nrfx/hal/nrf_radio.h>
+
+#include "radio_df.h"
+
+/* @brief Minimum antennae number required if antenna switching is enabled */
+#define DF_ANT_NUM_MIN 2
+/* @brief Value to set antenna GPIO pin as not connected */
+#define DF_GPIO_PIN_NOT_SET 0xFF
+/* @brief Number of PSEL_DFEGPIO registers in Radio peripheral */
+#define DF_PSEL_GPIO_NUM 8
+
+/* @brief Direction Finding antenna matrix configuration */
+struct df_ant_cfg {
+	uint8_t ant_num;
+	/* Maximum supported antenna switch pattern length */
+	uint8_t max_ant_sw_pattern_num;
+	/* Selection of GPIOs to be used to switch antennae by Radio */
+	uint8_t dfe_gpio[DF_PSEL_GPIO_NUM];
+};
+
+#define DFE_ANT DT_NODELABEL(dfe_ant)
+#define DFE_GPIO_PIN(idx)						\
+	UTIL_OR(UTIL_AND(DT_NODE_HAS_PROP(DFE_ANT, dfegpio##idx##_pin),	\
+				DT_PROP(DFE_ANT, dfegpio##idx##_pin)),	\
+		DF_GPIO_PIN_NOT_SET)
```

Comment:
This won't work for pin P0.00. Is that one not supposed to be ever used by this mechanism?

Reply:
Why it won't work? 
According to UTIL_AND documentation:

> Like **a && b**, but does evaluation and
> short-circuiting at C preprocessor time.
> 
> This is not the same as the binary &&, however; in particular,
> **a** should expand to an integer literal 0 or 1. However, **b**
> can be any value.

That means DT_PROP(DFE_ANT, dfegpio##idx##_pin) is allowed to be zero.
If there is dfe_gpio#_pin = <0>, zero should be returned by complete macro.

Reply:
But then you'll have `UTIL_OR(0, DF_GPIO_PIN_NOT_SET)`, right?

Reply:
You are right. I should use here COND_CODE_1 macro instead

Reply:
corrected